src/js/controller/tracks-loader.ts

```diff
@@ -96,8 +107,11 @@ function xhrSuccess(xhr, track, successHandler, errorHandler) {
     }
 }
 
-function loadVttParser() {
-    return require.ensure(['parsers/captions/vttparser'], function (require) {
-        return require('parsers/captions/vttparser').default;
-    }, chunkLoadWarningHandler(301131), 'vttparser');
+function loadVttParser(): any {
```

Comment:
```suggestion
function loadVttParser(): Promise<VTTParser> {
```

Reply:
without webpack-env, the name `require` isn't identified and it will throw an error. If it is included (see package.json changes), require.ensure is defined as returning void, so if I don't define as `any` the `then` method isn't supposed to exist.

Reply:
Did you try using `as` for the require.ensure statement?

How does package.json matter? I don’t know what changes you mean.

Reply:
Tech-debt for the near future: we should replace `require.ensure` code splitting imports with `import()`. Harmony syntax is supported by the latest versions of webpack and rollup.

Reply:
yeah, I tried that as well